ui/src/app/services/data/data.service.ts

```diff
@@ -0,0 +1,306 @@
+  private ddlSub = new BehaviorSubject({})
+  private sessionsSub = new BehaviorSubject({} as ISession[])
+  private configSub = new BehaviorSubject({} as ISpannerConfig)
+  // currentSessionSub not using any where
+  private currentSessionSub = new BehaviorSubject({} as ISession)
+  private isOfflineSub = new BehaviorSubject<boolean>(false)
+  private ruleMapSub = new BehaviorSubject<IRuleContent>({})
+
+  rule = this.ruleMapSub.asObservable().pipe(filter((res) => Object.keys(res).length !== 0))
+  conv = this.convSubject.asObservable().pipe(filter((res) => Object.keys(res).length !== 0))
+  conversionRate = this.conversionRateSub
+    .asObservable()
+    .pipe(filter((res) => Object.keys(res).length !== 0))
+  typeMap = this.typeMapSub.asObservable().pipe(filter((res) => Object.keys(res).length !== 0))
+  summary = this.summarySub.asObservable().pipe(filter((res) => res.size > 0))
+  ddl = this.ddlSub.asObservable().pipe(filter((res) => Object.keys(res).length !== 0))
+  sessions = this.sessionsSub.asObservable()
+  config = this.configSub.asObservable().pipe(filter((res) => Object.keys(res).length !== 0))
+  isOffline = this.isOfflineSub.asObservable()
+  currentSession = this.currentSessionSub
+    .asObservable()
+    .pipe(filter((res) => Object.keys(res).length !== 0))
+
+  constructor(private fetch: FetchService, private snackbar: SnackbarService) {
+    let inputType = localStorage.getItem(StorageKeys.Type) as string
+    let config: unknown = localStorage.getItem(StorageKeys.Config)
+
+    switch (inputType) {
+      case InputType.DirectConnect:
+        this.getSchemaConversionFromDb()
+        break
+
+      case InputType.DumpFile:
+        if (config !== null) {
+          this.getSchemaConversionFromDump(config as IDumpConfig)
+        }
+        break
+
+      case InputType.SessionFile:
+        if (config !== null) {
+          this.getSchemaConversionFromSession(config as ISessionConfig)
+        }
+        break
+
+      default:
+        console.log('not able to find input type')
+    }
+    this.getConfig()
+    this.updateIsOffline()
+  }
+
+  resetStore() {
+    this.convSubject.next({} as IConv)
+    this.conversionRateSub.next({})
+    this.typeMapSub.next({})
+    this.summarySub.next(new Map<string, ISummary>())
+    this.ddlSub.next({})
+  }
+
+  getDdl() {
+    this.fetch.getDdl().subscribe((res) => {
+      this.ddlSub.next(res)
+    })
+  }
+
+  getSchemaConversionFromDb() {
+    this.fetch.getSchemaConversionFromDirectConnect().subscribe((res: IConv) => {
+      this.convSubject.next(res)
+      // this.initiateSession()
+    })
+  }
+
+  //To do : remove snackbar call from dataservice
+  getAllSessions() {
+    this.fetch.getSessions().subscribe({
+      next: (sessions: ISession[]) => {
+        this.sessionsSub.next(sessions)
+      },
+      error: (err: any) => {
+        this.snackbar.openSnackBar('Not able to fetch Session..', 'Close', 5000)
```

Comment:
Replace the snackbars text 'Not able to' with 'Unable to ' for consistency in all snackbar msgs. Same for few more snackbars